src/user.jl

```diff
@@ -310,7 +310,26 @@ function preferunits(u0::Units, u::Units...)
             "For instance, it should not be used with units of dimension 𝐋^2.")
         end
         y = typeof(dim).parameters[1][1]
-        promotion[name(y)] = typeof(unit).parameters[1][1]
+        promotion[name(y)] = unit
+    end
+
+    # Let's run a quick check for anything where, for instance, current was defined is C/ms and time
+    # was defined in ns. We'll give a warning if this is the case so the user knows they may get
+    # unexpected (though still correct) results. There may be cases where people would want to do this,
+    # so we won't stop them, just let them know they're doing it.
+    ulist = [([typeof(i[2]).parameters[1] for i in promotion]...)...]
```

Comment:
I'm confused by this line:
```julia
[typeof(i[2]).parameters[1] for i in promotion]
```
is a vector generated by comprehension, then you splat it in a tuple, then you splat the tuple in a vector again?

Why not just
```suggestion
    ulist = (typeof(i[2]).parameters[1] for i in promotion)
```
which is a lazy generator instead?